### ISSUE TITLE
Unsafe Lifecycle warning in React >= 16.3

### ISSUE THREAD

**Issue by SebastianStamm** [snakesilk/react-fullscreen#7]
React 16.3.0 declares the `componentWillReceiveProps` lifecycle method as unsafe and shows warnings when using React.StrictMode

**Comment by pomle** [comment]
Thank you, I will try to find time to fix this!

**Comment by mreishus** [comment]
This should fix it.

Remove these lines:
```
    componentWillReceiveProps(nextProps) {
        this.handleProps(nextProps);
    }
```

Add these lines:
```
    componentDidUpdate(prevProps) {
        if (prevProps.enabled != this.props.enabled) {
            this.handleProps(this.props);
        }
    }
```

**Comment by mreishus** [comment]
sure, I can, give me a sec

**Comment by pomle** [comment]
@mreishus thanks, I'm already looking into it. Testing it right now.

**Comment by mreishus** [comment]
Ok, oops.  I didn't make a pull request initially because I was using a customized version of this in my project, which also had different spacing.. but I should have just gone ahead and done it, my bad.

**Comment by pomle** [comment]
@SebastianStamm, this has been fixed in `0.2.3`. Thanks for the report. Sorry for the slow response.
